deployments/stacks/azure-cluster.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository https://github.com/carbynestack/carbynestack.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as cdktf from "cdktf";
+import * as azurerm from "@cdktf/provider-azurerm";
+import { Construct } from "constructs";
+import { AzurePlatform } from "../constructs/platforms/azure-platform";
+import { CarbyneStack } from "../constructs/carbyne-stack";
+
+export interface AzureClusterConfig {
+  private: boolean;
+  jumpHostResourceGroup?: string;
+  jumpHostVirtualNetworkName?: string;
+}
+
+export default class AzureCluster extends cdktf.TerraformStack {
+  constructor(scope: Construct, id: string, config: AzureClusterConfig) {
+    super(scope, id);
+
+    new azurerm.provider.AzurermProvider(this, "AzureRm", {
+      features: {
+        resourceGroup: {
+          preventDeletionIfContainsResources: false,
+        },
+      },
+    });
+    // create a shared azure resource group for all the platforms
+    const azureResourceGroup = new azurerm.resourceGroup.ResourceGroup(
+      this,
+      "cs-caliper",
```

Comment:
is caliper the name to use?

Reply:
caliper name has been removed and replaced with generic `cs`